### PR TITLE
Feature/upgrade psalm

### LINKED ISSUE #28: Psalm violition
**Package version**: 1.5.2

**Description**
There are several issues found by Psalm when running on the main branch this will be an issue/annoyance when dealing with future PR's

*update*
This is only when using v5 of the package,  sorry I didn't notice sooner.

**To Reproduce**
- Run `./vendor/bin/psalm --no-progress`
- View output

**Additional Context**
```
./vendor/bin/psalm --no-progress

...

------------------------------
17 errors found
------------------------------
315 other issues found.
You can display them with --show-info=true
------------------------------

Checks took 0.05 seconds and used 5.589MB of memory
No files analyzed
Psalm was able to infer types for 91.2313% of the codebase
```

### PR DESCRIPTION
**Pull request by JimTools**
**Description**
Allowing Psalm v5 for versions of PHP which support it, is for maintenance to ensure the package stays up to date and fixing some potential issues which have been flagged as Psalm as it has gotten stricter

closes #28

### CODE REVIEW

File: src/core/Model/StackTraceInterface.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+declare(strict_types=1);
+
+namespace Scheb\Tombstone\Core\Model;
+
+/**
+ * @extends \IteratorAggregate<array-key, StackTraceFrame>
+ * @extends \ArrayAccess<array-key, StackTraceFrame>
+ */
+interface StackTraceInterface extends \IteratorAggregate, \ArrayAccess
```
Comment (JimTools) [inline src/core/Model/StackTraceInterface.php:11]: there is an odd with @impliments annotation I've had to extend the interface to satisfy error 

https://github.com/phpstan/phpstan/discussions/5484

File: composer.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
         "squizlabs/php_codesniffer": "^3.5",
         "symfony/css-selector": "^4.4|^5.0",
         "symfony/dom-crawler": "^4.4|^5.0",
-        "vimeo/psalm": "^3.12|^4.0"
+        "vimeo/psalm": "^3.12|^4.30|^5.22"
```
Comment (scheb) [inline composer.json]: I'd suggest we drop ^3.12 here, since it's quite out-of-date nowadays and PHP 7.1 support is also given with version 4.

**Review by scheb — COMMENTED** [review]
Looks good. Just one suggestion regarding Psalm version.

Also, would you please rebase your work onto the current main branch? I did some update so get the repository and builds up-to-date. You'll now get a proper CI build with Psalm 5.

**Comment by scheb** [comment]
Thanks, merged :)

**Pull request merged**